osu.Game/Screens/Select/SongSelect.cs

```diff
@@ -428,7 +428,7 @@ private void workingBeatmapChanged(ValueChangedEvent<WorkingBeatmap> e)
 
         private void updateSelectedBeatmap(BeatmapInfo beatmap)
         {
-            if (beatmap?.Equals(beatmapNoDebounce) == true)
+            if (beatmap == null || beatmap.Equals(beatmapNoDebounce))
```

Comment:
This is not the correct check for the problem I pointed out - it (correctly) fails other tests. Now if `beatmap` is null but `beatmapNoDebounce` wasn't, this early-returns instead of transferring, therefore breaking deselection. There was a reason I had it written as funny as it was.

Reply:
Yeah, I was thinking we wanted to keep the debounce from being null but didn't realise it isn't actually a bindable and null is valid in this context.